### PR TITLE
chore(install): report infra and logging recipes available

### PR DESCRIPTION
**Pull request by ctrombley**

### CODE REVIEW

File: internal/install/recipe_installer.go [hunk 1]
```diff
@@ -130,45 +140,40 @@ func (i *RecipeInstaller) Install() error {
...
 
-	log.Debugf(fmt.Sprintf("AssumeYes:%t.", i.AssumeYes))
-	log.Debugf(fmt.Sprintf("RecipeNames:%s.", i.RecipeNames))
-	log.Debugf(fmt.Sprintf("RecipePaths:%s.", i.RecipePaths))
-	log.Debugf(fmt.Sprintf("SkipDiscovery:%t.", i.SkipDiscovery))
-	log.Debugf(fmt.Sprintf("SkipInfraInstall:%t.", i.SkipInfraInstall))
-	log.Debugf(fmt.Sprintf("SkipLoggingInstall:%t.", i.SkipLoggingInstall))
-	log.Debugf(fmt.Sprintf("SkipIntegrations:%t.", i.SkipIntegrations))
-	log.Debugf(fmt.Sprintf("RecipesProvided:%t.", i.RecipesProvided()))
+	log.Debugf("InstallerContext: %+v\n", i.InstallerContext)
```
Comment (Julien4218) [inline internal/install/recipe_installer.go:149]: Can we keep the debug statement for the output of the method i.RecipesProvided() ?

File: internal/install/recipe_installer.go [hunk 2]
```diff
@@ -130,45 +140,40 @@ func (i *RecipeInstaller) Install() error {
 		}
 
 		for _, r := range recipes {
-			log.Debugf(fmt.Sprintf("Found available integration %s.", r.Name))
+			log.Debugf("Found available integration %s.", r.Name)
```
Comment (Julien4218) [inline internal/install/recipe_installer.go:143]: Do we need a line feed at the end?

File: internal/install/recipe_installer.go [hunk 3]
```diff
@@ -88,6 +88,16 @@ func (i *RecipeInstaller) Install() error {
 		return i.fail(err)
 	}
 
+	infraAgentRecipe, err := i.fetchRecipeAndReportAvailable(m, infraAgentRecipeName)
```
Comment (Julien4218) [inline internal/install/recipe_installer.go:91]: Is it relevant to fetch the infra and logging recipe when the `-c` flag is used?

File: internal/install/recipe_installer.go (hunk 2 above)
Comment (zlesnr) [reply internal/install/recipe_installer.go:143]: We shouldn't.  The `log` statement should do that for us.  If it were a `fmt.Sprintf`, then we would.

**Review by zlesnr — APPROVED** [review]
🌯

File: internal/install/recipe_installer.go (hunk 1 above)
Comment (ctrombley) [reply internal/install/recipe_installer.go:149]: 👍

File: internal/install/recipe_installer.go (hunk 3 above)
Comment (ctrombley) [reply internal/install/recipe_installer.go:91]: I think it is, since we want to show it as "Available" yet "Skipped" in the UI.

Comment (Julien4218) [reply internal/install/recipe_installer.go:91]: Sorry to insist @ctrombley but I don't think it's valid to fetch the infraAgentRecipe and loggingRecipe at this level.
For one, those are NOT used when a recipe is provided explicitly with the -c option, which is NOT the workflow of what a user would do from the NR UI.
Second, if we look at the usages of those 2 recipe objects, they are only used later down in the ShouldInstallInfraAgent( ) and ShouldInstallLogging( ) method, and symmetrically those are not used for the ShouldInstallIntegrations( ) path which is the way the recipe path provided execute (-c option).
Consequently, could we just move the fetch of infra/logging closer to the scope where they are used in their respective ShouldInstallXxx( ) methods?

**Review by Julien4218 — DISMISSED** [review]

**Comment by Julien4218** [comment]
Looking good!

**Review by Julien4218 — APPROVED** [review]

**Pull request merged**